cumulus/client/relay-chain-rpc-interface/src/metrics.rs

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+use prometheus::{Error as PrometheusError, HistogramTimer, Registry};
+use prometheus_endpoint::{HistogramOpts, HistogramVec, Opts};
+
+// Gathers metrics about the blockchain RPC client.
+#[derive(Clone)]
+pub(crate) struct BlockchainRpcMetrics {
+	rpc_request: HistogramVec,
+}
+
+impl BlockchainRpcMetrics {
+	pub(crate) fn register(registry: &Registry) -> Result<Self, PrometheusError> {
+		Ok(Self {
+			rpc_request: prometheus_endpoint::register(
+				HistogramVec::new(
+					HistogramOpts {
+						common_opts: Opts::new(
+							"polkadot_parachain_relay_chain_rpc_interface",
+							"Tracks stats about cumulus relay chain RPC interface",
+						),
+						buckets: prometheus::exponential_buckets(0.001, 4.0, 9)
```

Comment:
just curiosity - any reason for these values?

Am I right that buckets will be?
```
0.001
0.004
0.016
0.064
0.256
1.024
4.096
16.384
65.536
```

Reply:
Correct for the buckets.

I picked the buckets split by seeing it used for other requests timers in the code (related to substrate libp2p), although there isn't a particular relationship between them. Ideally we'll have some preliminary measurements for these first and then pick the buckets. I am ok with these values though because they correspond to some rough back of the envelope measurements for exchanging data over the network (e.g USA -> EU -> USA ~ 150 ms). I think that the higher buckets (e.g. >1s) can be considered extreme, and might correspond to super infrequent outliers (assuming the network runs fine most of the time).